test/SlowTests/Issues/RavenDB-17566.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Queries;
+using Raven.Client.Documents.Session;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Client.Documents.Conventions;
+using Xunit;
+using Xunit.Abstractions;
+
+
+namespace SlowTests.Issues;
+
+public class RavenDB_17566 : RavenTestBase
+{
+    public RavenDB_17566(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [Fact]
+    public async Task ProjectionOnNonStoredFieldsInIndex()
+    {
+        var myEntityIndex = new MyEntityIndex();
+        using var store = base.GetDocumentStore(new Options
+        {
+            ModifyDocumentStore = s => s.Conventions.FindProjectedPropertyNameForIndex =
+                (indexedType, indexedName, path, prop) =>
+                {
+                    if (indexedName == myEntityIndex.IndexName)
+                        return path + prop;
+
+                    return DocumentConventions.DefaultFindPropertyNameForIndex(indexedType, indexedName, path, prop);
```

Comment:
@ppekrol we want to share this method with users because it's more convenient to deal with conventions for them.